src/plugins/wizard/public/embeddable/wizard_component.tsx

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+import {
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiText,
+  EuiAvatar,
+  EuiFlexGrid,
+  EuiCodeBlock,
+} from '@elastic/eui';
+
+import { withEmbeddableSubscription } from '../../../embeddable/public';
+import { WizardEmbeddable, WizardInput, WizardOutput } from './wizard_embeddable';
+import { validateSchemaState } from '../application/utils/validate_schema_state';
+
+interface Props {
+  embeddable: WizardEmbeddable;
+  input: WizardInput;
+  output: WizardOutput;
+}
+
+function wrapSearchTerms(task?: string, search?: string) {
+  if (!search) return task;
+  if (!task) return task;
+  const parts = task.split(new RegExp(`(${search})`, 'g'));
+  return parts.map((part, i) =>
+    part === search ? (
+      <span key={i} style={{ backgroundColor: 'yellow' }}>
+        {part}
+      </span>
+    ) : (
+      part
+    )
+  );
+}
+
+function WizardEmbeddableComponentInner({
+  embeddable,
+  input: { search },
+  output: { savedAttributes },
+}: Props) {
+  const { ReactExpressionRenderer, toasts, types, indexPatterns, aggs } = embeddable;
+  const [expression, setExpression] = useState<string>();
+  const { title, description, visualizationState, styleState } = savedAttributes || {};
+
+  useEffect(() => {
```

Comment:
A lot of logic here seems to overlap with the existing D&D workspace code. Can we consolidate them instead of maintaining it in two places?